dnsapi/dns_schlundtech.sh

```diff
@@ -0,0 +1,200 @@
+# Usage: dns_schlundtech_add _acme-challenge.www.domain.com "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+
+dns_schlundtech_add() {
+  fulldomain="$1"
+  txtvalue="$2"
+
+  _SLTEC_credentials
+  if [ "$?" -ne 0 ]; then
+    _err "Please specify the SchlundTech user and password and try again."
+    return 1
+  fi
+
+  _SLTEC_split_domain "$fulldomain"
+
+  _info "Using the schlundtech dns api to set the ${fulldomain} record"
+  _debug "fulldomain: ${fulldomain}"
+  _debug "domain    : ${_SLTEC_domain}"
+  _debug "subdomain : ${_SLTEC_subdomain}"
+  _debug "txtvalue  : ${txtvalue}"
+
+  _SLTEC_init_request_add "$SLTEC_user" "$SLTEC_password" "$SLTEC_context" "$_SLTEC_domain" "$_SLTEC_subdomain" "$txtvalue"
+  _debug "xmladd: ${_SLTEC_xmladd}"
+
+  _SLTEC_send_request "$_SLTEC_xmladd" "$SLTEC_server"
+  echo "$_SLTEC_response" | grep "<code>S0202</code>" >/dev/null
+  _ST_add_result="$?"
+  _debug "result: ${_ST_add_result}"
+
+  # returns 0 means success, otherwise error.
+  return "$_ST_add_result"
+}
+
+# Remove the txt record after validation.
+# Usage: dns_schlundtech_rm _acme-challenge.www.domain.com "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+
+dns_schlundtech_rm() {
+  fulldomain="$1"
+  txtvalue="$2"
+
+  _SLTEC_credentials
+  if [ "$?" -ne 0 ]; then
+    _err "Please specify the SchlundTech user and password and try again."
+    return 1
+  fi
+
+  _SLTEC_split_domain "$fulldomain"
+
+  _info "Using the schlundtech dns api to remove the ${fulldomain} record"
+  _debug "fulldomain: ${fulldomain}"
+  _debug "txtvalue  : ${txtvalue}"
+  _debug "domain    : ${_SLTEC_domain}"
+  _debug "subdomain : ${_SLTEC_subdomain}"
+
+  _SLTEC_init_request_rm "$SLTEC_user" "$SLTEC_password" "$SLTEC_context" "$_SLTEC_domain" "$_SLTEC_subdomain" "$txtvalue"
+  _debug "xmlrm:  ${_SLTEC_xmlrm}"
+
+  _SLTEC_send_request "$_SLTEC_xmlrm" "$SLTEC_server"
+  echo "$_SLTEC_response" | grep "<code>S0202</code>" >/dev/null
+  _ST_rm_result="$?"
+  _debug "result: ${_ST_rm_result}"
+
+  # no return value documented
+  #return "$_ST_rm_result"
+}
+
+####################  private functions below ##################################
+
+_SLTEC_credentials() {
+
+  if [ -z "${SLTEC_context}" ]; then
+    SLTEC_context="${SLTEC_context_default}"
+  fi
+
+  if [ -z "${SLTEC_server}" ]; then
+    SLTEC_server="${SLTEC_server_default}"
+  fi
+
+  if [ -z "${SLTEC_user}" ] || [ -z "$SLTEC_password" ] || [ -z "${SLTEC_context}" ] || [ -z "${SLTEC_server}" ]; then
+    SLTEC_user=""
+    SLTEC_password=""
+    SLTEC_context=""
+    SLTEC_server=""
+    return 1
+  else
+    _saveaccountconf SLTEC_user "${SLTEC_user}"
+    _saveaccountconf SLTEC_password "${SLTEC_password}"
+    _saveaccountconf SLTEC_context "${SLTEC_context}"
+    _saveaccountconf SLTEC_server "${SLTEC_server}"
+    return 0
+  fi
+}
+
+_SLTEC_split_domain() {
+  _ST_split_fulldomain="$1"
+
+  _SLTEC_domain="$(echo "$_ST_split_fulldomain" | sed 's/.*\.\([^.]*\.[^.]*\)/\1/')"
```

Comment:
do not just split the domain.
please use the api of Schlundtech to get a domain list, then check which part is the root zone.

otherwise you will get confused by `example.uk`  and `example.co.uk`

Reply:
Yep, you're right. I'll check into this,

Reply:
here is a guide for you:  https://github.com/Neilpang/acme.sh/wiki/DNS-API-Dev-Guide